### ISSUE TITLE
Nullable enum serialized as `text NOT NULL`

### ISSUE THREAD

**Issue by lukehutch** [serverpod/serverpod#1645]
**Describe the bug**

I have a protocol class with a nullable enum field:

```yaml
  vipStatus: VipStatus?
```

This is what `serverpod generate` produces, in `generated/tables.sql`:

```sql
  "vipStatus" text NOT NULL,
```

Serverpod git latest version.

**Comment by Isakdl** [comment]
`generated/tables.sql` is deprecated and removed (we do not generate it on main anymore). Do you have the same problem with the `definition.sql`?

**Comment by lukehutch** [comment]
I'm running a fork of main (rebased last night), and `generated/tables.sql` was not removed by `serverpod generate`. It's still there. Furthermore, `generated/migration/migrations/definition.sql` does not include the tables defined in `generated/tables.sql`. I need both, or my database is missing tables when I create a server.

**Comment by lukehutch** [comment]
In fact even after `serverpod generate`, the only place that column is defined is `generated/tables.sql`. From within my server project:

```bash
$ grep -r '"vipStatus" text' .
./generated/tables.pgsql:  "vipStatus" text NOT NULL,
```

And I am running the git latest version of `serverpod generate`, straight from the git repo:

```bash
dart ../vendor/serverpod/tools/serverpod_cli/bin/serverpod_cli.dart generate --watch
```

**Comment by lukehutch** [comment]
Maybe this is only being set up properly by `serverpod create`, not by `serverpod generate`? That's the only thing I can think of...

**Comment by lukehutch** [comment]
OK, I see your other comment on this over at #1419... I'll move the discussion from there to here, since that issue is closed.

I'll delete all my comments above, because the missing link was the new migration documentation.

Running `serverpod create-migration` was the thing I was missing. However, that failed the first time:

```console
Unable to determine latest database definition due to a corrupted migration. Please re-create or remove the migration version and try again. Migration version: "serverpod_chat".
PathNotFoundException: Cannot open file, path = '/home/luke/Work/click/click_server/generated/migration/migrations/serverpod_chat/definition_project.json' (OS Error: No such file or directory, errno = 2)
✗
```

I had to `rm generated/migration -r` to fix it.

I will close this, but wanted to alert you to the above issue. Thanks!

**Comment by Isakdl** [comment]
But anyway this is a different issue than the one opened here, did you experience the problem reported here with the new migration system?

**Comment by lukehutch** [comment]
No, everything worked fine once I migrated properly. It just wasn't at all obvious that I needed to do that, because I was running `serverpod generate`, destroying the Docker volumes and recreating them, and tables I needed were not all being generated. I hadn't seen the new documentation yet.

> But yes perhaps we could be a little smarter and check if the db state is completely clean and if so run the migration.

From an external view, this is the only thing that would make any sense. Really.

I'll create a new issue for that. Thanks!
